infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/DeserializableStringMapTypeDefinition.java

```diff
@@ -49,6 +50,10 @@ public Map<String, String> deserialize(final JsonParser p) throws IOException {
     for (; t == JsonToken.FIELD_NAME; t = p.nextToken()) {
       final String fieldName = p.getCurrentName();
       p.nextToken();
+      if (!p.getCurrentToken().isScalarValue()) {
+        throw MismatchedInputException.from(
```

Comment:
MismatchedInputException actually looks like the right type for this - it's a more explicit subtype of JsonProcessingException.

It probably doesn't matter but I'd be tempted to pass `String.class` as the expected class here.

Reply:
Okay cool. I will give that a shot. This was just a copy paste of:

https://github.com/ConsenSys/teku/blob/a4620f9598db35252261130d9fa2fe925b52ea3f/infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/DeserializableArrayTypeDefinition.java#L51-L52

Reply:
Ah yeah, in that case it's less clear what the class we need is (it's an array of something by TypeDefinition doesn't really know).  Whereas in this case we know we want a string.  It's a minor detail though - not sure it even turns up in the error message anywhere.